File: src/main/java/ru/trukhmanov/twochairsbackend/entity/DeckQuestion.java
package ru.trukhmanov.twochairsbackend.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name="deck_questions")
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
public class DeckQuestion {

    @EmbeddedId
    private DeckQuestionId id;

    public static DeckQuestion of(long deckId, long questionId) {
        return DeckQuestion.builder()
                .id(new DeckQuestionId(deckId, questionId))
                .build();
    }
}

File: src/main/kotlin/ru/trukhmanov/twochairsbackend/entity/DeckQuestion.kt
package ru.trukhmanov.twochairsbackend.entity

import jakarta.persistence.EmbeddedId
import jakarta.persistence.Entity
import jakarta.persistence.Table

@Entity
@Table(name = "deck_questions")
class DeckQuestion(
    @EmbeddedId
    var id: DeckQuestionId = DeckQuestionId()
) {
    companion object {
        fun of(deckId: Long, questionId: Long): DeckQuestion {
            return DeckQuestion(DeckQuestionId(deckId, questionId))
        }
    }
}
